pkg/core/xds/testdata/rules/02.golden.yaml

```diff
@@ -0,0 +1,103 @@
+- Conf:
+    action: ALLOW
+  Origin: null
+  Subset:
+  - Key: env
+    Not: false
+    Value: dev
+  - Key: version
+    Not: false
+    Value: v1
+  - Key: zone
+    Not: false
+    Value: us-east
+- Conf:
+    action: DENY
+  Origin: null
+  Subset:
+  - Key: env
+    Not: true
+    Value: dev
+  - Key: version
+    Not: false
+    Value: v1
+  - Key: zone
+    Not: false
+    Value: us-east
+- Conf:
+    action: ALLOW
+  Origin: null
+  Subset:
+  - Key: env
+    Not: false
+    Value: dev
+  - Key: version
+    Not: false
+    Value: v1
+  - Key: zone
+    Not: true
+    Value: us-east
+- Conf:
+    action: ALLOW
+  Origin: null
+  Subset:
+  - Key: env
+    Not: false
+    Value: dev
+  - Key: version
+    Not: true
+    Value: v1
+  - Key: zone
+    Not: false
+    Value: us-east
+- Conf:
+    action: DENY
+  Origin: null
+  Subset:
+  - Key: env
+    Not: true
+    Value: dev
+  - Key: version
+    Not: true
+    Value: v1
+  - Key: zone
+    Not: false
+    Value: us-east
+- Conf:
+    action: ALLOW
+  Origin: null
+  Subset:
+  - Key: env
+    Not: true
+    Value: dev
+  - Key: version
+    Not: false
+    Value: v1
+  - Key: zone
+    Not: true
+    Value: us-east
+- Conf:
+    action: ALLOW
+  Origin: null
+  Subset:
+  - Key: env
+    Not: false
+    Value: dev
+  - Key: version
+    Not: true
+    Value: v1
+  - Key: zone
+    Not: true
+    Value: us-east
+- Conf: null
```

Comment:
This is odd no?

Reply:
This is an indication for the XDS generator that policies don't specify behavior for this specific subset. Depending on the generator this should be treated differently, for example, the MeshTrafficPermission generator treats this as `action: DENY`

Reply:
Yes, it's equivalent for MeshTrafficPermission, but just because we generate "ALLOW" list. If you don't have policies for the subset it doesn't necessarily mean you don't have to send conf to the envoy.

Reply:
But isn't this just "fallback to default" it's just like not having this conf in the first place no?

Reply:
> But isn't this just "fallback to default" it's just like not having this conf in the first place no?

It is, unless XDS generator iterates over rules and generates special Envoy configs when `conf: nil`. I can't tell now if there is a policy that requires it. So probably we can delete rules that have `conf: nil`

Reply:
fixed, now the rule won't be added if `conf` is `nil`